addons/website_sale/static/src/snippets/s_dynamic_category_snippet/options.js

```diff
@@ -0,0 +1,109 @@
+        this.categories = await this.orm.call("product.public.category", "get_snippet_categories", []);
+        return _super(...arguments);
+    },
+
+    /**
+     *
+     * @override
+     */
+    async onBuilt() {
+        this.$target.get(0).dataset['columns'] = 2;
+        this.$target.get(0).dataset['height'] = "small";
+        this.$target.get(0).dataset['filterId'] = 0;
+        this.$target.get(0).dataset['alignment'] = "left";
+        this.$target.get(0).dataset['button'] = "Explore Now";
+    },
+
+    /**
+     *
+     * @override
+     * @private
+     */
+    _renderCustomXML: async function (uiFragment) {
+        const filtersSelectorEl = uiFragment.querySelector("[data-name='filter_opt']");
+        for (let index in this.categories) {
+            const category = this.categories[index]
+            const button = document.createElement("we-button");
+            button.dataset.selectDataAttribute = category['id'];
+            button.innerText = category['name'];
+            filtersSelectorEl.appendChild(button);
+        }
+    },
+})
+
+options.registry.dynamic_snippet_category_item = productRibbonMixin(options.Class.extend({
+    /**
+     * @override
+     */
+    willStart: async function () {
+        this.recordId = parseInt(this.$target.get(0).dataset.categoryId);
+        this.recordModel = 'product.public.category';
+        return this._super(...arguments);
+    },
+
+    onSetRibbon: async function(previewMode) {
+        if(!previewMode){
+            this.trigger_up('request_save', {reload: true, optionSelector: '.s_dynamic_category'});
+        }
+    }
+}))
+
+class categoryMedia extends MediaDialog{
+    async save(){
+        const nodeData = this.props.node.parentElement.dataset
+        rpc('/snippets/category/set_image',{
+            category_id: parseInt(nodeData.categoryId),
+            media: this.selectedMedia[this.state.activeTab],
+        })
+        this.props.close()
+        await super.save();
+    }
+}
+
+options.registry.ReplaceMedia.include({
+    async replaceMedia() {
+        if (this.$target.closest(".s_dynamic_category").length > 0) {
+            const imageEl = this.$target.get(0).querySelector(".s_category_image")
+            if(this.$target.get(0).classList.contains("category_item")){
+                this.call('dialog', 'add', categoryMedia, {
+                    node: imageEl,
+                    noDocuments: true,
+                    noVideos: true,
+                    noIcons: true,
+                    save:() => {this.trigger_up('request_save', {
+                        reload: true, optionSelector: ".s_dynamic_category"
+                    })},
+                })
+            }else{
+                await this.options.wysiwyg.openMediaDialog({
+                    node: imageEl,
+                    noDocuments: true,
+                    noVideos: true,
+                    noIcons: true,
+                });
+            }
+        }else {
```

Comment:
```suggestion
        } else {
```

similar for other files as well